webpack/assets/javascripts/react_app/components/statistics/ChartBox.js

```diff
@@ -29,30 +31,29 @@ class ChartBox extends React.Component {
   }
 
   render() {
-    const { chart } = this.props;
-
-    const modalConfig = getLargePieChartConfig({
-      data: this.props.chart.data,
-      id: `${chart.id}Modal`,
-    });
-
+    const { chart, type } = this.props;
+    const components = {
+      donut: DonutChart,
+    };
+    const Chart = components[type];
+    const isExist = () => chart.data && chart.data.length > 0;
     const tooltip = {
-      onClick: this.onClick,
+      onClick: isExist() ? this.onClick : null,
       title: this.props.tip,
       'data-toggle': 'tooltip',
       'data-placement': 'top',
+      className: isExist() ? 'pointer' : '',
```

Comment:
shouldn't we ignore all those props when we have no data?

Reply:
we still need those props, in order to render a chartbox, the inside chart would be empty, however the wrapping frame shouldn't.